src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -69,6 +70,17 @@ def list(self):
             osd['stats'] = {}
             osd['stats_history'] = {}
             osd_spec = str(osd_id)
+            metadata = str(mgr._ceph_get_metadata('osd', osd_spec))  # pylint: disable = W0212
+            device_ids = metadata.find('device_ids')
+            device_paths = metadata.find('device_paths')
+            devices = metadata[device_ids:device_paths-3]
+            devices = devices[14:len(devices)-1]
+            if devices == "":
```

Comment:
Following this: https://docs.python.org/3/library/stdtypes.html#truth-value-testing
You can simplify it:
```
osd['device_count'] = 0
if devices:
    device_list = devices.split(',')
    osd['device_count'] = len(device_list)
```